src/java.base/share/classes/java/util/zip/ZipCoder.java

```diff
@@ -224,8 +224,13 @@ int normalizedHash(byte[] a, int off, int len) {
             while (off < end) {
                 byte b = a[off];
                 if (b < 0) {
-                    // Non-ASCII, fall back to decoder loop
-                    return normalizedHashDecode(h, a, off, end);
+                    // Non-ASCII, fall back to decoding a String
+                    String s = JLA.newStringUTF8NoRepl(a, end - len, end);
+                    h = s.hashCode();
```

Comment:
Would it be possible to call normalizedHash(String name) here?

Reply:
Yes, updated. (The PR was supposed to be a draft until now since I was fixing up some testing issues with my first stab at this.)